client/dashboard/sites/views.ts

```diff
@@ -271,18 +271,27 @@ export function recordViewChanges(
 /**
  * Sanitize the view preference data by removing any invalid or malformed entries.
  */
-function sanitizeView( { sort, ...view }: SitesView ) {
-	if ( sort?.field ) {
-		const field = DEFAULT_FIELDS.find( ( field ) => field.id === sort?.field );
+function sanitizeView( view: SitesView ) {
+	// If no sanitization is needed then a reference to the same object should be returned.
+	let sanitized = view;
+
+	if ( sanitized.type === 'grid' && sanitized.layout?.previewSize ) {
+		// From PreviewSizePicker imageSizes in GB https://github.com/WordPress/gutenberg/blob/58a5abc7714bdff204d5f6bc350980f73686d54f/packages/dataviews/src/dataviews-layouts/grid/preview-size-picker.tsx#L14
+		const validSizes = [ 120, 170, 230, 290, 350, 430 ];
+		if ( ! validSizes.includes( sanitized.layout.previewSize ) ) {
+			sanitized.layout = { ...sanitized.layout, previewSize: 230 };
```

Comment:
Actually, this is a lot more straight forward 😅 
```suggestion
			sanitized.layout.previewSize = 230;
```

Reply:
Actually maybe I should set this to `undefined` 🤔 that way it is reverting back to the default.

Reply:
I decided to `delete` the invalid `previewSize` settings after all. Sorry for the churn @arthur791004 😅